platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileManager.java

```diff
@@ -143,6 +153,8 @@ public SignedStateFileManager(
                     savedStates[savedStates.length - 1].getMetadata().minimumGenerationNonAncient();
             if (generationNonAncient != null) {
                 minimumGenerationNonAncientForOldestState = generationNonAncient;
+                minimumGenerationNonAncientConsumer.newMinimumGenerationNonAncient(
```

Comment:
instead of doing this, why don't we just listen for an event where a signed state is stored to disk

Reply:
This needs to be called when we delete the oldest state from disk, not when we write a new state.